src/main/java/com/epam/test_generator/dao/impl/CaseVersionDAOImpl.java

```diff
@@ -66,4 +92,106 @@ public void delete(Iterable<Case> cases) {
             delete(caze);
         }
     }
+
+    private List<CaseVersion> getCaseVersions(List<Change> changes) {
+        TreeMap<CommitMetadata, List<Change>> changesWithSameCommitId =
+            getSortedGroupedMapByChangesWithSameCommitId(changes);
+
+        List<CaseVersion> caseVersions = new ArrayList<>(changesWithSameCommitId.size());
+        changesWithSameCommitId.forEach(
+            (commitMetadata, changeList) -> {
+                CaseVersion caseVersion = new CaseVersion();
+
+                caseVersion.setAuthor(commitMetadata.getAuthor());
+                caseVersion.setCommitId(commitMetadata.getId().toString());
+                caseVersion.setUpdatedDate(Date.from(commitMetadata.getCommitDate().atZone(
+                    ZoneId.systemDefault()).toInstant()));
+                caseVersion.setPropertyDifferences(getPropertyDifferences(changeList));
+
+                caseVersions.add(caseVersion);
+            }
+        );
+        return caseVersions;
+    }
+
+    private List<PropertyDifference> getPropertyDifferences(List<Change> changeList) {
+        List<PropertyDifference> propertyDifferences = new ArrayList<>();
+        for (Change change : changeList) {
+            if (change instanceof PropertyChange) {
+                if (change instanceof ValueChange) {
```

Comment:
Multiple if statements is better than nested if statement is not huge.

Reply:
done